### PR TITLE
refactor: Use cmp instead of msgpack-c for events packing.

### PR DESCRIPTION
**Pull request by iphydf**
This library is better in every way.

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/2175)

### CODE REVIEW

**Review by JFreegman — APPROVED** [review]
> Reviewed 31 of 31 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/2175)* status: :shipit: complete! 1 of 1 approvals obtained

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "third_party/cmp"]
+	path = third_party/cmp
+	url = https://github.com/camgunz/cmp
```
Comment (Green-Sky) [inline .gitmodules:3]: maybe make it shallow?

Comment (iphydf) [reply .gitmodules]: I'd rather leave that up to the cloner.

File: CMakeLists.txt [hunk 2]
```diff
@@ -145,6 +145,8 @@ if(BUILD_TOXAV)
   endif()
 endif()
 
+add_definitions(-DCMP_NO_FLOAT=1)
```
Comment (sudden6) [inline CMakeLists.txt:148]: Please add a  comment what this does/why it's needed.

Comment (iphydf) [reply CMakeLists.txt:150]: Done.

**Review by sudden6 — APPROVED** [review]

**Review by Green-Sky — APPROVED** [review]
hack to make reviewable happy, maybe

**Pull request merged**